Documentation/ContentBlocks/Carousel.rst

```diff
@@ -0,0 +1,182 @@
+:composer:`friendsoftypo3/content-blocks`. We assume that you are already
+familiar with the concepts described in chapter
+`Custom Content Blocks <https://docs.typo3.org/permalink/t3sitepackage:content-blocks>`_.
+
+..  contents:: Table of contents
+
+..  _carousel-directory:
+
+Directory structure of the carousel content element
+===================================================
+
+Additionally to the files that the jumbotron (compare
+:ref:`content-blocks-jumbotron-directory`) the carousel comes with special CSS
+and JavaScript needed for this element only.
+
+Additionally it supplies a template for its display in the backend.
+
+..  directory-tree::
+    :level: 2
+    :show-file-icons: true
+
+    *   packages/my_site_package/ContentBlocks/ContentElements/jumbotron
+
+        *   assets
+
+            *   frontend.css
+            *   frontend.js
+            *   icon.svg
+
+        *   language
+
+            *   labels.xlf
+
+        *   templates
+
+            *   backend-preview.html
+            *   frontend.html
+
+        *   config.yaml
+
+..  _carousel-configuration:
+
+The carousel: A Content Block containing a collection of elements
+=================================================================
+
+The carousel can contain one or several items, each of which has an image,
+header, and description:
+
+..  literalinclude:: /CodeSnippets/my_site_package/ContentBlocks/ContentElements/carousel/config.yaml
+    :caption: packages/my_site_package/ContentBlocks/ContentElements/carousel/config.yaml
+    :linenos:
+
+Line 8: We use a field of type
+`Collection <https://docs.typo3.org/permalink/friendsoftypo3/content-blocks:field-type-collection>`_
+to contain the items to be displayed in the carousel. This field type expects
+an array of fields (line 10ff).
+
+Line 15: We use the type
+`File <https://docs.typo3.org/permalink/friendsoftypo3/content-blocks:field-type-file>`_
+to reference the image for the carousel item. We allow images only (line 16)
+and require exactly one image (lines 17 and 18).
+
+Line 20: The title should be one line of text. We use the type
+`Text <https://docs.typo3.org/permalink/friendsoftypo3/content-blocks:field-type-text>`_.
+
+Line 22: The description may contain rich text. Therefore, we use the type
+`Textarea <https://docs.typo3.org/permalink/friendsoftypo3/content-blocks:field-type-textarea>`_
+and enable the Rich-Text Editor (line 23).
+
+..  _carousel-fluid:
+
+Frontend template: Fluid template for a Content Block with a Collection
+=======================================================================
+
+..  literalinclude:: /CodeSnippets/my_site_package/ContentBlocks/ContentElements/carousel/templates/frontend.html
+    :caption: packages/my_site_package/ContentBlocks/ContentElements/carousel/templates/frontend.html
+    :linenos:
+
+Line 1: We use `Asset.css ViewHelper <f:asset.css> <https://docs.typo3.org/permalink/t3viewhelper:typo3-fluid-asset-css>`_
```

Comment:
Where is the `cb` ViewHelper defined?

Reply:
It doesnt have docu yet, once documented we can link there